# init 
importing libs

In [1]:
import numpy as np
import pandas as pd
import matplotlib as plt

# Reading_cvs

In [2]:
df = pd.read_csv('../data/athletes_sport_details.csv')
country_df = pd.read_csv('../data/country_ranks.csv')
df.head()

ID                      Name Sex   Age  Height  Weight         Team  NOC  \
0   1                 A Dijiang   M  24.0   180.0    80.0        China  CHN   
1   2                  A Lamusi   M  23.0   170.0    60.0        China  CHN   
2   3       Gunnar Nielsen Aaby   M  24.0   191.0    85.0      Denmark  DEN   
3   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  Netherlands  NED   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  Netherlands  NED   

         Games  Year  Season       City          Sport  \
0  1992 Summer  1992  Summer  Barcelona     Basketball   
1  2012 Summer  2012  Summer     London           Judo   
2  1920 Summer  1920  Summer  Antwerpen       Football   
3  1988 Winter  1988  Winter    Calgary  Speed Skating   
4  1988 Winter  1988  Winter    Calgary  Speed Skating   

                                Event Medal  Sport_Rank  Has_Previous_Medal  
0         Basketball Men's Basketball   NaN           4               False  
1        Judo Men's Extra-Lightweight   NaN          18               False  
2             Football Men's Football   NaN          12               False  
3    Speed Skating Women's 500 metres   NaN          20               False  
4  Speed Skating Women's 1,000 metres   NaN          20               False

# Making a model just for Gymnastics athletics

In [3]:
Gymnastics_df = df[
    (df['Sport'] == 'Gymnastics') &
    (df['Event'].isin([
        "Gymnastics Women's Individual All-Around",
        "Gymnastics Men's Individual All-Around"
    ]))
]

Gymnastics_df.head() , Gymnastics_df.describe()

(      ID                     Name Sex   Age  Height  Weight          Team  \
 32    17  Paavo Johannes Aaltonen   M  28.0   175.0    64.0       Finland   
 40    17  Paavo Johannes Aaltonen   M  32.0   175.0    64.0       Finland   
 107   51        Nstor Abad Sanjun   M  23.0   167.0    64.0         Spain   
 298  244         Raouf Abdelraouf   M  22.0   167.0    63.0         Egypt   
 474  394               Irene Abel   F  19.0   160.0    48.0  East Germany   
 
      NOC        Games  Year  Season            City       Sport  \
 32   FIN  1948 Summer  1948  Summer          London  Gymnastics   
 40   FIN  1952 Summer  1952  Summer        Helsinki  Gymnastics   
 107  ESP  2016 Summer  2016  Summer  Rio de Janeiro  Gymnastics   
 298  EGY  2000 Summer  2000  Summer          Sydney  Gymnastics   
 474  GDR  1972 Summer  1972  Summer          Munich  Gymnastics   
 
                                         Event   Medal  Sport_Rank  \
 32     Gymnastics Men's Individual All-Around  Br

In [4]:
Gymnastics_df_5000_sample = Gymnastics_df[Gymnastics_df["Year"] >= 2000]
Gymnastics_df_5000_sample

ID                           Name Sex   Age  Height  Weight  \
107         51              Nstor Abad Sanjun   M  23.0   167.0    64.0   
298        244               Raouf Abdelraouf   M  22.0   167.0    63.0   
485        396                     Katja Abel   F  25.0   165.0    55.0   
823        627       Andreea Roxana Acatrinei   F  16.0   150.0    40.0   
1311       976         Jonna Eva-Maj Adlerteg   F  17.0   153.0    38.0   
...        ...                            ...  ..   ...     ...     ...   
205061  134974  Nataliya Kamilovna Ziganshina   F  18.0   156.0    47.0   
205517  135228            Epke Jan Zonderland   M  22.0   173.0    69.0   
205600  135269                        Zou Kai   M  20.0   158.0    55.0   
205645  135291   Roman Volodymyrovych Zozulia   M  21.0   164.0    59.0   
205653  135291   Roman Volodymyrovych Zozulia   M  25.0   164.0    59.0   

               Team  NOC        Games  Year  Season            City  \
107           Spain  ESP  2016 Summer  2016  Summer  Rio de Janeiro   
298           Egypt  EGY  2000 Summer  2000  Summer          Sydney   
485         Germany  GER  2008 Summer  2008  Summer         Beijing   
823         Romania  ROU  2008 Summer  2008  Summer         Beijing   
1311         Sweden  SWE  2012 Summer  2012  Summer          London   
...             ...  ...          ...   ...     ...             ...   
205061       Russia  RUS  2004 Summer  2004  Summer          Athina   
205517  Netherlands  NED  2008 Summer  2008  Summer         Beijing   
205600        China  CHN  2008 Summer  2008  Summer         Beijing   
205645      Ukraine  UKR  2000 Summer  2000  Summer          Sydney   
205653      Ukraine  UKR  2004 Summer  2004  Summer          Athina   

             Sport                                     Event Medal  \
107     Gymnastics    Gymnastics Men's Individual All-Around   NaN   
298     Gymnastics    Gymnastics Men's Individual All-Around   NaN   
485     Gymnastics  Gymnastics Women's Individual All-Around   NaN   
823     Gymnastics  Gymnastics Women's Individual All-Around   NaN   
1311    Gymnastics  Gymnastics Women's Individual All-Around   NaN   
...            ...                                       ...   ...   
205061  Gymnastics  Gymnastics Women's Individual All-Around   NaN   
205517  Gymnastics    Gymnastics Men's Individual All-Around   NaN   
205600  Gymnastics    Gymnastics Men's Individual All-Around   NaN   
205645  Gymnastics    Gymnastics Men's Individual All-Around   NaN   
205653  Gymnastics    Gymnastics Men's Individual All-Around   NaN   

        Sport_Rank  Has_Previous_Medal  
107             14               False  
298             14               False  
485             14               False  
823             14               False  
1311            14               False  
...            ...                 ...  
205061          14               False  
205517          14               False  
205600          14               False  
205645          14               False  
205653          14                True  

[777 rows x 17 columns]

In [5]:
# Dictionary to store the last cumulative count per name
last_count = {}

# Initialize the column
Gymnastics_df_5000_sample['Count'] = 0

# Sort by Name and Year to ensure proper ordering
Gymnastics_df_5000_sample = Gymnastics_df_5000_sample.sort_values(['Name', 'Year']).reset_index(drop=True)

# Iterate through rows
for name, group in Gymnastics_df_5000_sample.groupby('Name', sort=False):
    # Keep track of cumulative count for this name
    count = 0
    for year, year_group in group.groupby('Year', sort=False):
        # Assign the current count to all rows of this year
        indices = year_group.index
        Gymnastics_df_5000_sample.loc[indices, 'Count'] = count
        # Update cumulative count for next year
        count += len(year_group)

Gymnastics_df_5000_sample = Gymnastics_df_5000_sample.sort_values(by='ID').reset_index(drop=True)
Gymnastics_df_5000_sample

C:\Users\aamyr\AppData\Local\Temp\ipykernel_17076\2131119701.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Gymnastics_df_5000_sample['Count'] = 0


ID                           Name Sex   Age  Height  Weight  \
0        51              Nstor Abad Sanjun   M  23.0   167.0    64.0   
1       244               Raouf Abdelraouf   M  22.0   167.0    63.0   
2       396                     Katja Abel   F  25.0   165.0    55.0   
3       627       Andreea Roxana Acatrinei   F  16.0   150.0    40.0   
4       976         Jonna Eva-Maj Adlerteg   F  17.0   153.0    38.0   
..      ...                            ...  ..   ...     ...     ...   
772  134974  Nataliya Kamilovna Ziganshina   F  18.0   156.0    47.0   
773  135228            Epke Jan Zonderland   M  22.0   173.0    69.0   
774  135269                        Zou Kai   M  20.0   158.0    55.0   
775  135291   Roman Volodymyrovych Zozulia   M  21.0   164.0    59.0   
776  135291   Roman Volodymyrovych Zozulia   M  25.0   164.0    59.0   

            Team  NOC        Games  Year  Season            City       Sport  \
0          Spain  ESP  2016 Summer  2016  Summer  Rio de Janeiro  Gymnastics   
1          Egypt  EGY  2000 Summer  2000  Summer          Sydney  Gymnastics   
2        Germany  GER  2008 Summer  2008  Summer         Beijing  Gymnastics   
3        Romania  ROU  2008 Summer  2008  Summer         Beijing  Gymnastics   
4         Sweden  SWE  2012 Summer  2012  Summer          London  Gymnastics   
..           ...  ...          ...   ...     ...             ...         ...   
772       Russia  RUS  2004 Summer  2004  Summer          Athina  Gymnastics   
773  Netherlands  NED  2008 Summer  2008  Summer         Beijing  Gymnastics   
774        China  CHN  2008 Summer  2008  Summer         Beijing  Gymnastics   
775      Ukraine  UKR  2000 Summer  2000  Summer          Sydney  Gymnastics   
776      Ukraine  UKR  2004 Summer  2004  Summer          Athina  Gymnastics   

                                        Event Medal  Sport_Rank  \
0      Gymnastics Men's Individual All-Around   NaN          14   
1      Gymnastics Men's Individual All-Around   NaN          14   
2    Gymnastics Women's Individual All-Around   NaN          14   
3    Gymnastics Women's Individual All-Around   NaN          14   
4    Gymnastics Women's Individual All-Around   NaN          14   
..                                        ...   ...         ...   
772  Gymnastics Women's Individual All-Around   NaN          14   
773    Gymnastics Men's Individual All-Around   NaN          14   
774    Gymnastics Men's Individual All-Around   NaN          14   
775    Gymnastics Men's Individual All-Around   NaN          14   
776    Gymnastics Men's Individual All-Around   NaN          14   

     Has_Previous_Medal  Count  
0                 False      0  
1                 False      0  
2                 False      0  
3                 False      0  
4                 False      0  
..                  ...    ...  
772               False      0  
773               False      0  
774               False      0  
775               False      0  
776                True      1  

[777 rows x 18 columns]

making ranking for every country based on economy and technology

In [6]:
x=Gymnastics_df_5000_sample["Team"].unique()
len(x)

73

In [7]:
result = pd.merge(Gymnastics_df_5000_sample, country_df, how="left", on=["Team"])
result = result.drop(columns=["Sport_Rank","Year"])
Gymnastics_df_5000_sample = result[["ID","Name","Sex","Age","Height","Weight","Team","Development_Status","Rank","NOC","Games","Season","City","Sport","Event","Has_Previous_Medal","Count","Medal"]]

In [8]:
Gymnastics_df_5000_sample = Gymnastics_df_5000_sample.dropna(subset=["Development_Status", "Rank"])
Gymnastics_df_5000_sample["ID"] = range(1, len(Gymnastics_df_5000_sample) + 1)
Gymnastics_df_5000_sample.to_csv('../data/Gymnastics_sample.csv')
Gymnastics_df_5000_sample

ID                           Name Sex   Age  Height  Weight  \
0      1              Nstor Abad Sanjun   M  23.0   167.0    64.0   
1      2               Raouf Abdelraouf   M  22.0   167.0    63.0   
2      3                     Katja Abel   F  25.0   165.0    55.0   
3      4       Andreea Roxana Acatrinei   F  16.0   150.0    40.0   
4      5         Jonna Eva-Maj Adlerteg   F  17.0   153.0    38.0   
..   ...                            ...  ..   ...     ...     ...   
772  773  Nataliya Kamilovna Ziganshina   F  18.0   156.0    47.0   
773  774            Epke Jan Zonderland   M  22.0   173.0    69.0   
774  775                        Zou Kai   M  20.0   158.0    55.0   
775  776   Roman Volodymyrovych Zozulia   M  21.0   164.0    59.0   
776  777   Roman Volodymyrovych Zozulia   M  25.0   164.0    59.0   

            Team Development_Status  Rank  NOC        Games  Season  \
0          Spain          Developed     1  ESP  2016 Summer  Summer   
1          Egypt         Developing     3  EGY  2000 Summer  Summer   
2        Germany          Developed     1  GER  2008 Summer  Summer   
3        Romania          Developed     1  ROU  2008 Summer  Summer   
4         Sweden          Developed     1  SWE  2012 Summer  Summer   
..           ...                ...   ...  ...          ...     ...   
772       Russia       Upper-Middle     2  RUS  2004 Summer  Summer   
773  Netherlands          Developed     1  NED  2008 Summer  Summer   
774        China       Upper-Middle     2  CHN  2008 Summer  Summer   
775      Ukraine         Developing     3  UKR  2000 Summer  Summer   
776      Ukraine         Developing     3  UKR  2004 Summer  Summer   

               City       Sport                                     Event  \
0    Rio de Janeiro  Gymnastics    Gymnastics Men's Individual All-Around   
1            Sydney  Gymnastics    Gymnastics Men's Individual All-Around   
2           Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
3           Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
4            London  Gymnastics  Gymnastics Women's Individual All-Around   
..              ...         ...                                       ...   
772          Athina  Gymnastics  Gymnastics Women's Individual All-Around   
773         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
774         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
775          Sydney  Gymnastics    Gymnastics Men's Individual All-Around   
776          Athina  Gymnastics    Gymnastics Men's Individual All-Around   

     Has_Previous_Medal  Count Medal  
0                 False      0   NaN  
1                 False      0   NaN  
2                 False      0   NaN  
3                 False      0   NaN  
4                 False      0   NaN  
..                  ...    ...   ...  
772               False      0   NaN  
773               False      0   NaN  
774               False      0   NaN  
775               False      0   NaN  
776                True      1   NaN  

[777 rows x 18 columns]

- we gather some data now lets shift to other processes
- endcoding

In [9]:
Gymnastics_df_5000_sample["Sex"] = Gymnastics_df_5000_sample["Sex"].map({"M": 0, "F": 1})
Gymnastics_df_5000_sample["Has_Previous_Medal"] = Gymnastics_df_5000_sample["Has_Previous_Medal"].map({False: 0, True: 1})
Gymnastics_df_5000_sample["Rank"] = Gymnastics_df_5000_sample["Development_Status"].map({
    "Developed": 3,
    "Upper-Middle": 2,
    "Developing": 1
})

In [10]:
Gymnastics_df_5000_sample.head()

ID                      Name  Sex   Age  Height  Weight     Team  \
0   1         Nstor Abad Sanjun    0  23.0   167.0    64.0    Spain   
1   2          Raouf Abdelraouf    0  22.0   167.0    63.0    Egypt   
2   3                Katja Abel    1  25.0   165.0    55.0  Germany   
3   4  Andreea Roxana Acatrinei    1  16.0   150.0    40.0  Romania   
4   5    Jonna Eva-Maj Adlerteg    1  17.0   153.0    38.0   Sweden   

  Development_Status  Rank  NOC        Games  Season            City  \
0          Developed     3  ESP  2016 Summer  Summer  Rio de Janeiro   
1         Developing     1  EGY  2000 Summer  Summer          Sydney   
2          Developed     3  GER  2008 Summer  Summer         Beijing   
3          Developed     3  ROU  2008 Summer  Summer         Beijing   
4          Developed     3  SWE  2012 Summer  Summer          London   

        Sport                                     Event  Has_Previous_Medal  \
0  Gymnastics    Gymnastics Men's Individual All-Around                   0   
1  Gymnastics    Gymnastics Men's Individual All-Around                   0   
2  Gymnastics  Gymnastics Women's Individual All-Around                   0   
3  Gymnastics  Gymnastics Women's Individual All-Around                   0   
4  Gymnastics  Gymnastics Women's Individual All-Around                   0   

   Count Medal  
0      0   NaN  
1      0   NaN  
2      0   NaN  
3      0   NaN  
4      0   NaN

- i use Z-score formula for standardizition (scaling)
- x′=(x−μ)/σ

In [11]:
num_cols = ["Age", "Height", "Weight","Count","Rank","Has_Previous_Medal"]
for col in num_cols:
    mean = Gymnastics_df_5000_sample[col].mean()
    std = Gymnastics_df_5000_sample[col].std()
    Gymnastics_df_5000_sample[col] = (Gymnastics_df_5000_sample[col] - mean) / std
Gymnastics_df_5000_sample = Gymnastics_df_5000_sample.drop(columns=["ID"])
Gymnastics_df_5000_sample.to_csv('../data/Scaled_gymnastics_sample.csv')
Gymnastics_df_5000_sample.describe()

Sex           Age        Height        Weight          Rank  \
count  777.000000  7.770000e+02  7.770000e+02  7.770000e+02  7.770000e+02   
mean     0.513514  2.926302e-16  1.579746e-15 -1.371704e-16  1.463151e-16   
std      0.500139  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   
min      0.000000 -1.730595e+00 -3.069027e+00 -2.490500e+00 -2.294109e+00   
25%      0.000000 -7.590107e-01 -7.039890e-01 -7.832936e-01 -8.245933e-01   
50%      1.000000 -3.032292e-02  1.969778e-01  7.030966e-02  6.449227e-01   
75%      1.000000  6.983649e-01  7.600820e-01  7.342233e-01  6.449227e-01   
max      1.000000  3.370220e+00  2.674637e+00  3.105343e+00  6.449227e-01   

       Has_Previous_Medal       Count  
count        7.770000e+02  777.000000  
mean         7.315755e-17    0.000000  
std          1.000000e+00    1.000000  
min         -3.044791e-01   -0.471164  
25%         -3.044791e-01   -0.471164  
50%         -3.044791e-01   -0.471164  
75%         -3.044791e-01   -0.471164  
max          3.280071e+00    5.530389

- we check Std to be near 1 

In [12]:
Gymnastics_df_5000_sample

Name  Sex       Age    Height    Weight  \
0                Nstor Abad Sanjun    0  0.455469  0.760082  0.923913   
1                 Raouf Abdelraouf    0  0.212573  0.760082  0.829068   
2                       Katja Abel    1  0.941261  0.534840  0.070310   
3         Andreea Roxana Acatrinei    1 -1.244803 -1.154472 -1.352362   
4           Jonna Eva-Maj Adlerteg    1 -1.001907 -0.816610 -1.542052   
..                             ...  ...       ...       ...       ...   
772  Nataliya Kamilovna Ziganshina    1 -0.759011 -0.478747 -0.688449   
773            Epke Jan Zonderland    0  0.212573  1.435807  1.398137   
774                        Zou Kai    0 -0.273219 -0.253506  0.070310   
775   Roman Volodymyrovych Zozulia    0 -0.030323  0.422219  0.449689   
776   Roman Volodymyrovych Zozulia    0  0.941261  0.422219  0.449689   

            Team Development_Status      Rank  NOC        Games  Season  \
0          Spain          Developed  0.644923  ESP  2016 Summer  Summer   
1          Egypt         Developing -2.294109  EGY  2000 Summer  Summer   
2        Germany          Developed  0.644923  GER  2008 Summer  Summer   
3        Romania          Developed  0.644923  ROU  2008 Summer  Summer   
4         Sweden          Developed  0.644923  SWE  2012 Summer  Summer   
..           ...                ...       ...  ...          ...     ...   
772       Russia       Upper-Middle -0.824593  RUS  2004 Summer  Summer   
773  Netherlands          Developed  0.644923  NED  2008 Summer  Summer   
774        China       Upper-Middle -0.824593  CHN  2008 Summer  Summer   
775      Ukraine         Developing -2.294109  UKR  2000 Summer  Summer   
776      Ukraine         Developing -2.294109  UKR  2004 Summer  Summer   

               City       Sport                                     Event  \
0    Rio de Janeiro  Gymnastics    Gymnastics Men's Individual All-Around   
1            Sydney  Gymnastics    Gymnastics Men's Individual All-Around   
2           Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
3           Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
4            London  Gymnastics  Gymnastics Women's Individual All-Around   
..              ...         ...                                       ...   
772          Athina  Gymnastics  Gymnastics Women's Individual All-Around   
773         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
774         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
775          Sydney  Gymnastics    Gymnastics Men's Individual All-Around   
776          Athina  Gymnastics    Gymnastics Men's Individual All-Around   

     Has_Previous_Medal     Count Medal  
0             -0.304479 -0.471164   NaN  
1             -0.304479 -0.471164   NaN  
2             -0.304479 -0.471164   NaN  
3             -0.304479 -0.471164   NaN  
4             -0.304479 -0.471164   NaN  
..                  ...       ...   ...  
772           -0.304479 -0.471164   NaN  
773           -0.304479 -0.471164   NaN  
774           -0.304479 -0.471164   NaN  
775           -0.304479 -0.471164   NaN  
776            3.280071  1.529353   NaN  

[777 rows x 17 columns]

In [13]:
print("isnull count:")
print(Gymnastics_df_5000_sample.isna().sum())

isnull count:
Name                    0
Sex                     0
Age                     0
Height                  0
Weight                  0
Team                    0
Development_Status      0
Rank                    0
NOC                     0
Games                   0
Season                  0
City                    0
Sport                   0
Event                   0
Has_Previous_Medal      0
Count                   0
Medal                 747
dtype: int64


- data is extreme imbalance due to every year only gets 3 medals and there is only 2 sex (5*3)*2 = 30
- and there is 777 - 747 = 30 so we should oversample or undersample them

- due to this data i prefer to use oversample to make this extreme imbalance faide

In [16]:
medal = Gymnastics_df_5000_sample[Gymnastics_df_5000_sample["Medal"].notna()]
no_medal = Gymnastics_df_5000_sample[Gymnastics_df_5000_sample["Medal"].isna()]

medal_oversampled = medal.sample(len(no_medal), replace=True, random_state=42)
df_balanced = pd.concat([medal_oversampled, no_medal])
df_balanced.to_csv('../data/Generated_data.csv')
df_balanced

Name  Sex       Age    Height    Weight  \
295   Shawn Machel Johnson (-East)    1 -1.244803 -1.717577 -1.257518   
552   Alexandra Rose "Aly" Raisman    1  0.212573 -0.366126 -0.214225   
744                     Yang Yilin    1 -1.487699 -1.154472 -1.826586   
475    Aliya Farkhatovna Mustafina    1 -1.001907  0.196978 -0.593604   
382    Danell Johan Leyva Gonzalez    0 -0.273219  1.097945  1.303292   
..                             ...  ...       ...       ...       ...   
772  Nataliya Kamilovna Ziganshina    1 -0.759011 -0.478747 -0.688449   
773            Epke Jan Zonderland    0  0.212573  1.435807  1.398137   
774                        Zou Kai    0 -0.273219 -0.253506  0.070310   
775   Roman Volodymyrovych Zozulia    0 -0.030323  0.422219  0.449689   
776   Roman Volodymyrovych Zozulia    0  0.941261  0.422219  0.449689   

              Team Development_Status      Rank  NOC        Games  Season  \
295  United States          Developed  0.644923  USA  2008 Summer  Summer   
552  United States          Developed  0.644923  USA  2016 Summer  Summer   
744          China       Upper-Middle -0.824593  CHN  2008 Summer  Summer   
475         Russia       Upper-Middle -0.824593  RUS  2012 Summer  Summer   
382  United States          Developed  0.644923  USA  2012 Summer  Summer   
..             ...                ...       ...  ...          ...     ...   
772         Russia       Upper-Middle -0.824593  RUS  2004 Summer  Summer   
773    Netherlands          Developed  0.644923  NED  2008 Summer  Summer   
774          China       Upper-Middle -0.824593  CHN  2008 Summer  Summer   
775        Ukraine         Developing -2.294109  UKR  2000 Summer  Summer   
776        Ukraine         Developing -2.294109  UKR  2004 Summer  Summer   

               City       Sport                                     Event  \
295         Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
552  Rio de Janeiro  Gymnastics  Gymnastics Women's Individual All-Around   
744         Beijing  Gymnastics  Gymnastics Women's Individual All-Around   
475          London  Gymnastics  Gymnastics Women's Individual All-Around   
382          London  Gymnastics    Gymnastics Men's Individual All-Around   
..              ...         ...                                       ...   
772          Athina  Gymnastics  Gymnastics Women's Individual All-Around   
773         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
774         Beijing  Gymnastics    Gymnastics Men's Individual All-Around   
775          Sydney  Gymnastics    Gymnastics Men's Individual All-Around   
776          Athina  Gymnastics    Gymnastics Men's Individual All-Around   

     Has_Previous_Medal     Count   Medal  
295           -0.304479 -0.471164  Silver  
552            3.280071  1.529353  Silver  
744           -0.304479 -0.471164  Bronze  
475           -0.304479 -0.471164  Bronze  
382           -0.304479 -0.471164  Bronze  
..                  ...       ...     ...  
772           -0.304479 -0.471164     NaN  
773           -0.304479 -0.471164     NaN  
774           -0.304479 -0.471164     NaN  
775           -0.304479 -0.471164     NaN  
776            3.280071  1.529353     NaN  

[1494 rows x 17 columns]